File: cases/ProjetoTestC/src/router.c
/*
 * Escolhe o tratador adequado para cada requisição.
 *
 * Elo do caminho vulnerável: router_dispatch -> handler_execute.
 *
 * O ramo alternativo, handle_unknown, existe para que a busca em
 * profundidade precise descartar caminhos que não levam à função
 * vulnerável.
 */

#include <string.h>

#include "../include/handler.h"
#include "../include/logger.h"
#include "../include/router.h"

/* Trata requisições que não correspondem a nenhuma rota conhecida.
 *
 * Não leva à função vulnerável. */
static void handle_unknown(const char *request)
{
    log_error("Rota desconhecida.");

    (void)request;
}

void router_dispatch(const char *request)
{
    log_info("Requisicao recebida.");

    if (strncmp(request, "RUN ", 4) == 0) {
        handler_execute(request + 4);
    } else {
        handle_unknown(request);
    }
}

File: cases/ProjetoTestPython/src/router.py
"""Escolhe o tratador adequado para cada requisição.

Elo do caminho vulnerável: router_dispatch -> handler_execute.

O ramo alternativo, handle_unknown, existe para que a busca em
profundidade precise descartar caminhos que não levam à função
vulnerável.
"""

from handler import handler_execute
from logger import log_error, log_info


def handle_unknown(request):
    """Trata requisições que não correspondem a nenhuma rota conhecida.

    Não leva à função vulnerável.
    """
    log_error("Rota desconhecida.")


def router_dispatch(request):
    """Encaminha a requisição conforme o seu prefixo."""
    log_info("Requisicao recebida.")

    if request.startswith("RUN "):
        return handler_execute(request[4:])

    return handle_unknown(request)
